### ISSUE TITLE
Change streamer for SvtTrack to keep backward compatible with older data.

### ISSUE THREAD

**Issue by mholtrop** [JeffersonLab/hps-dst#50]
Issue #35 changed the data store structure of the SvtTrack class but did not create a custom streamer to allow it to still read older DST data. 
A custom _read_ streamer will determine data type and read data accordingly.

**Comment by mholtrop** [comment]
Turns out this is a non-issue for ROOT v5 and v6 IF the ROOT dictionaries are created with a + in the LinkDef file. In that case ROOT is supposed to take care of the schema evolution automatically. See: https://root.cern.ch/root/SchemaEvolution.html
I just verified that this indeed works by making a root file from an slcio file using the old dst-maker, and then again with the new dst-maker. The root file from the old dst-maker can be read with the new libHpsEvent, however if you request:
SvtTrack *st2 = evt2->getTrack(0)
st2->getGblTrack()
You are given a nullptr. 

The only object that does not have the + is HpsMCParticle.  That + should be added.
I will do this on this branch and test for consistency.
